### PR TITLE
X967 unit test added for slotmapper component

### PR DESCRIPTION
**Pull request by seenanair**

### CODE REVIEW

File: src/components/labware/Labware.tsx [hunk 1]
```diff
@@ -144,13 +144,15 @@ const Labware = ({
   slotBuilder
 }: React.PropsWithChildren<LabwareProps>) => {
   const labwareMachine = React.useMemo(() => {
-    return createLabwareMachine({
+    return createLabwareMachine();
+  }, []);
+  const [current, send] = useMachine(labwareMachine, {
```
Comment (khelwood) [inline src/components/labware/Labware.tsx:149]: Why has this context parameter been introduced?

File: src/components/labware/labware.machine.ts [hunk 2]
```diff
@@ -1,34 +1,21 @@
...
-  SelectionMode
-} from './labware.types';
+import { LabwareMachineContext, LabwareMachineEvent, LabwareMachineSchema } from './labware.types';
 import { assign } from '@xstate/immer';
 import { pure } from 'xstate/lib/actions';
-import { SlotFieldsFragment } from '../../types/sdk';
 import { emptySlots, filledSlots, findSlotByAddress, isSlotEmpty, isSlotFilled } from '../../lib/helpers/slotHelper';
 import { sortDownRight } from '../../lib/helpers/labwareHelper';
 
-interface CreateLabwareMachineParams {
```
Comment (khelwood) [inline src/components/labware/labware.machine.ts:15]: Why has this interface been deleted?

File: src/components/labware/Labware.tsx (hunk 1 above)
Comment (seenanair) [reply src/components/labware/Labware.tsx:149]: There was a warning before in console 'Machine given to `useMachine` has changed between renders. This is not supported and might lead to unexpected results.'
So, if we pass in parameters while creating  machine based on a component prop, machine instance will be recreated whenever this prop/parameter changes and there by changing the useMachine which accept this machine instance as param which is not recommended. This  way of creating and using the  state machine , in which we create a static instance and then defining the context of state machine using the passed in parameters which will ensure that the machine instance will not be changed.

File: src/components/labware/labware.machine.ts (hunk 2 above)
Comment (seenanair) [reply src/components/labware/labware.machine.ts:15]: This interface was to define the structure of params for createLabwareMachine which we are not using any more

**Review by khelwood — APPROVED** [review]

**Pull request merged**